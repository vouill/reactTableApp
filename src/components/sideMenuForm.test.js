import React from 'react'
import {shallow, mount} from 'enzyme';
import SideMenuFormComponent from './sideMenuForm.component'

describe('SideMenuFormComponent', () => {
    let Component
    let ComponentWithChildren
    const onUrlInput = jest.fn()
    beforeEach(() => {
        Component = shallow(
            <SideMenuFormComponent
                onUrlInput={onUrlInput}
            ></SideMenuFormComponent>
        );
        ComponentWithChildren = mount(
            <SideMenuFormComponent
                onUrlInput={onUrlInput}
            ></SideMenuFormComponent>
        );
    });
    it('should render', () => {
        expect(ComponentWithChildren.find('label')).toHaveLength(1)
         expect(Component).toMatchSnapshot();
    })

})