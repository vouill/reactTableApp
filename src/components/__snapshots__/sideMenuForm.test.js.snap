exports[`SideMenuFormComponent should render 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <div>
        <ControlLabel
            bsClass="control-label"
            srOnly={false}>
            Github event stream url
        </ControlLabel>
        <form
            onSubmit={[Function]}>
            <input
                className="form-group form-control"
                onChange={[Function]}
                placeholder="Event stream url"
                required={true}
                type="url" />
            <ButtonToolbar
                bsClass="btn-toolbar">
                <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="default"
                    disabled={false}
                    type="submit">
                     Go !
                </Button>
                <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="success"
                    disabled={false}
                    onClick={[Function]}>
                    Rails
                </Button>
                <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="success"
                    className="ngrxButton"
                    disabled={false}
                    onClick={[Function]}>
                    Ngrx
                </Button>
            </ButtonToolbar>
        </form>
    </div>
</div>,
  "nodes": Array [
    <div>
      <div>
            <ControlLabel
                  bsClass="control-label"
                  srOnly={false}>
                  Github event stream url
            </ControlLabel>
            <form
                  onSubmit={[Function]}>
                  <input
                        className="form-group form-control"
                        onChange={[Function]}
                        placeholder="Event stream url"
                        required={true}
                        type="url" />
                  <ButtonToolbar
                        bsClass="btn-toolbar">
                        <Button
                              active={false}
                              block={false}
                              bsClass="btn"
                              bsStyle="default"
                              disabled={false}
                              type="submit">
                               Go !
                        </Button>
                        <Button
                              active={false}
                              block={false}
                              bsClass="btn"
                              bsStyle="success"
                              disabled={false}
                              onClick={[Function]}>
                              Rails
                        </Button>
                        <Button
                              active={false}
                              block={false}
                              bsClass="btn"
                              bsStyle="success"
                              className="ngrxButton"
                              disabled={false}
                              onClick={[Function]}>
                              Ngrx
                        </Button>
                  </ButtonToolbar>
            </form>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <SideMenuFormComponent
        onUrlInput={[Function]} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": SideMenuFormComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleUrlButtonClick": [Function],
        "handleUrlChange": [Function],
        "props": Object {
          "onUrlInput": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <div>
                    <ControlLabel
                              bsClass="control-label"
                              srOnly={false}>
                              Github event stream url
                    </ControlLabel>
                    <form
                              onSubmit={[Function]}>
                              <input
                                        className="form-group form-control"
                                        onChange={[Function]}
                                        placeholder="Event stream url"
                                        required={true}
                                        type="url" />
                              <ButtonToolbar
                                        bsClass="btn-toolbar">
                                        <Button
                                                  active={false}
                                                  block={false}
                                                  bsClass="btn"
                                                  bsStyle="default"
                                                  disabled={false}
                                                  type="submit">
                                                   Go !
                                        </Button>
                                        <Button
                                                  active={false}
                                                  block={false}
                                                  bsClass="btn"
                                                  bsStyle="success"
                                                  disabled={false}
                                                  onClick={[Function]}>
                                                  Rails
                                        </Button>
                                        <Button
                                                  active={false}
                                                  block={false}
                                                  bsClass="btn"
                                                  bsStyle="success"
                                                  className="ngrxButton"
                                                  disabled={false}
                                                  onClick={[Function]}>
                                                  Ngrx
                                        </Button>
                              </ButtonToolbar>
                    </form>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <div>
                    <ControlLabel
                              bsClass="control-label"
                              srOnly={false}>
                              Github event stream url
                    </ControlLabel>
                    <form
                              onSubmit={[Function]}>
                              <input
                                        className="form-group form-control"
                                        onChange={[Function]}
                                        placeholder="Event stream url"
                                        required={true}
                                        type="url" />
                              <ButtonToolbar
                                        bsClass="btn-toolbar">
                                        <Button
                                                  active={false}
                                                  block={false}
                                                  bsClass="btn"
                                                  bsStyle="default"
                                                  disabled={false}
                                                  type="submit">
                                                   Go !
                                        </Button>
                                        <Button
                                                  active={false}
                                                  block={false}
                                                  bsClass="btn"
                                                  bsStyle="success"
                                                  disabled={false}
                                                  onClick={[Function]}>
                                                  Rails
                                        </Button>
                                        <Button
                                                  active={false}
                                                  block={false}
                                                  bsClass="btn"
                                                  bsStyle="success"
                                                  className="ngrxButton"
                                                  disabled={false}
                                                  onClick={[Function]}>
                                                  Ngrx
                                        </Button>
                              </ButtonToolbar>
                    </form>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SideMenuFormComponent
    onUrlInput={[Function]} />,
}
`;
